js/hw7.js

```diff
@@ -0,0 +1,108 @@
+// task 1
+const playList = [
+    {
+        author: "LED ZEPPELIN",
+        song: "STAIRWAY TO HEAVEN"
+    },
+    {
+        author: "QUEEN",
+        song: "BOHEMIAN RHAPSODY"
+    },
+    {
+        author: "LYNYRD SKYNYRD",
+        song: "FREE BIRD"
+    },
+    {
+        author: "DEEP PURPLE",
+        song: "SMOKE ON THE WATER"
+    },
+    {
+        author: "JIMI HENDRIX",
+        song: "ALL ALONG THE WATCHTOWER"
+    },
+    {
+        author: "AC/DC",
+        song: "BACK IN BLACK"
+    },
+    {
+        author: "QUEEN",
+        song: "WE WILL ROCK YOU"
+    },
+    {
+        author: "METALLICA",
+        song: "ENTER SANDMAN"
+    }
+];
+
+function printPlayList() {
+    let list = document.createElement("OL");
+    list.setAttribute("id", "playlist");
+    document.body.appendChild(list);
+
+    let sections = document.getElementsByTagName('section');
+    let color = window.getComputedStyle(sections[0]).backgroundColor;
+    list.style.backgroundColor = color;
+    list.style.margin = 0;
+    list.style.padding = '10px 10px 30px 30px';
+
+    playList.forEach(element => {
+        var y = document.createElement("LI");
+        var t = document.createTextNode(`${element.author} - ${element.song}`);
+        y.appendChild(t);
+        document.getElementById("playlist").appendChild(y);
+    });
+}
+
+// task 2
+function showModalWindow() {
+    var modal = document.getElementById("modal");
+    var btn = document.getElementById("openModal");
+    var closeBtn = document.getElementById("closeModal");
+
+    btn.onclick = function () {
+        modal.style.display = "block";
+    }
+
+    closeBtn.onclick = function () {
+        modal.style.display = "none";
+    }
+
+    window.onclick = function (event) {
+        if (event.target == modal) {
+            modal.style.display = "none";
+        }
+    }
+}
+
+// task3 
+let lights;
+let stage = 0;
+
+function initLights() {
+    lights = document.getElementsByClassName('light');
+}
+
+function switchColor() {
+    switch (stage % 4) {
+        case 0:
+            lights[0].style.background = 'red';
```

Comment:
Good solution. But try with class toggling. If the element will have class active - it will lighten up an if is disabled - default background - blak

Reply:
transition should be 'red' - 'red' 'yellow' - 'green' -  'yellow' - 'red' and in 'yellow' part i was not able not find out how manage it with only toggle.